src/command_modules/azure-cli-sf/azure/cli/command_modules/sf/custom.py

```diff
@@ -153,6 +153,14 @@ def sf_select(endpoint, cert=None,
     set_global_config_value("servicefabric", "endpoint", endpoint)
 
 
+def validate_app_path(app_path):
+    abspath = os.path.abspath(app_path)
+    if os.path.isdir(abspath):
+        return abspath
+    else:
+        raise CLIError("Invalid path to application directory: {0}".format(abspath))
```

Comment:
I would suggest a `ValueError` to be throw instead. The error here is not CLI specific but input parameter issue.